crates/storage/provider/src/providers/state/historical.rs

```diff
@@ -96,26 +102,32 @@ impl<'a, 'b, TX: DbTx<'a>> StateProvider for HistoricalStateProviderRef<'a, 'b,
         // history key to search IntegerList of transition id changesets.
         let history_key = StorageShardedKey::new(address, storage_key, self.transition);
 
-        let Some(changeset_transition_id) = self.tx.cursor_read::<tables::StorageHistory>()?
+        let changeset_transition_id = self
+            .tx
+            .cursor_read::<tables::StorageHistory>()?
             .seek(history_key)?
-            .filter(|(key,_)| key.address == address && key.sharded_key.key == storage_key)
-            .map(|(_,list)| list.0.enable_rank().successor(self.transition as usize).map(|i| i as u64)) else {
-                return Ok(None)
-            };
+            .filter(|(key, _)| key.address == address && key.sharded_key.key == storage_key)
+            .map(|(_, list)| {
+                list.0.enable_rank().successor(self.transition as usize).map(|i| i as u64)
+            });
```

Comment:
We should move this to a helper, but OK for now (cc @rakita for when we do the provider refactor)